improver/utilities/mathematical_operations.py

```diff
@@ -404,3 +406,236 @@ def fast_linear_fit(
+
+    @staticmethod
+    def calculate_anomalies(diagnostic_cube, mean_cube, variance_cube=None) -> ndarray:
+        """Calculate anomalies from the input cubes."""
+        anomalies_data = diagnostic_cube.data - mean_cube.data
+        if variance_cube:
+            anomalies_data = anomalies_data / np.sqrt(variance_cube.data)
+        else:
+            pass
+        return anomalies_data
+
+    @staticmethod
+    def _generate_output_name_and_units(diagnostic_cube, variance_cube=None) -> str:
+        """Get suitable output name and units from input cube metadata"""
+        anomaly_cube = diagnostic_cube.copy()
+        new_name = None
+        new_units = "None"
+        # If a variance cube is provided, the output is a standard anomaly and units are removed
+        if variance_cube: 
+            new_units = ""
+            try:
+                anomaly_cube.standard_name = (
+                    anomaly_cube.standard_name + "_standard_anomaly"
+                )
+                new_name = anomaly_cube.standard_name
+            except ValueError:
+                anomaly_cube.long_name = (
+                    anomaly_cube.standard_name + "_standard_anomaly"
+                )
+                new_name = anomaly_cube.long_name
+        else:
+            new_units = anomaly_cube.units
+            try:
+                anomaly_cube.standard_name = anomaly_cube.standard_name + "_anomaly"
+                new_name = anomaly_cube.standard_name
+            except ValueError:
+                anomaly_cube.long_name = anomaly_cube.standard_name + "_anomaly"
+                new_name = anomaly_cube.long_name 
+            variance_cube is not None
+            
+        return new_name, new_units
+
+    @staticmethod
+    def _add_reference_epoch_metadata(output_cube, diagnostic_cube, mean_cube):
+        """Add epoch metadata to describe the creation of the anomaly in place.
+        1. Add a scalar coordinate called 'reference epoch' with (1) timebound from mean cube for anomaly cube,
+        and (2) validity time from diagnostic cube. This is to describe the time period over which the climatology was calculated.
+        2. Add a cell method called 'anomaly' to describe the operation that was performed. cell_method = reference epoch: anomaly.
+        """
+        reference_epoch = iris.coords.AuxCoord(
+            points=mean_cube.coord("time").points,
+            bounds=mean_cube.coord("time").bounds
+            if diagnostic_cube.coord("time").has_bounds()
+            else None,
+            long_name="reference_epoch",
+            units=mean_cube.coord("time").units,
+        )
+
+        output_cube.add_aux_coord(reference_epoch)
+
+        cell_method = CellMethod(
+            method="anomaly", coords="reference_epoch"
+        )  # Create a cell method to describe the operation that was performed
+        output_cube.add_cell_method(cell_method)
+
+        return output_cube
+
+    @staticmethod
```

Comment:
```suggestion
    @staticmethod
```

This can be removed once you start using `self` within the function